python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -140,10 +141,51 @@ def execute_action(self, data):
                 return
 
         project_entity, entities = self._get_entities_from_payload(data)
+        config_entity = data["pc"]
+        command_name = data["name"]
+
+        # We have to do a couple of things here. The first is that we ALWAYS
+        # stick to the non-legacy code path for the __core_info and __upgrade_check
+        # commands. Neither of these require a bootstrap to occur, and are therefore
+        # fast when run through the modern code path. In addition, the implementation
+        # of these "special" project-level commands already outputs markdown syntax,
+        # which is preferable to the HTML we would get from the tank command.
+        #
+        # The second bit is that we can identify commands that need to be run in
+        # legacy mode by whether the config entity dictionary we got from the client
+        # contains a "_legacy_config_root" key that was shoved into it when the
+        # get_actions method ran through its own legacy code path. In that case, we
+        # shell out to the tank command by way of our process_manager object instead
+        # of using the modern code path.
+        if command_name not in ["__core_info", "__upgrade_check"]:
+            if "_legacy_config_root" in config_entity:
+                (out, err, retcode) = self.process_manager.execute_toolkit_command(
```

Comment:
If I understand correctly this means that we're now a bit faster than v1. This looks like we'll now only get actions for entities that the browser asks us for instead of simply scanning every shotgun_*.yml files like before. 

Correct?